clients/rospy/src/rospy/core.py

```diff
@@ -223,6 +223,9 @@ def __call__(self, caller_id, period):
               (now - last_logging_time) > rospy.Duration(period)):
             self.last_logging_time_table[caller_id] = now
             return True
+        elif (last_logging_time > now):
+            self.last_logging_time_table = {}
```

Comment:
After resetting the dictionary shouldn't the `caller_id` still be added: `self.last_logging_time_table[caller_id] = now`?

Reply:
Yes, this should prevent losing a log message when the time goes back

Reply:
I don't think it would have lost a message but always also printed the next message. Anyway its fixes. Thanks.